dask/utils.py

```diff
@@ -1111,6 +1111,17 @@ def partial_by_order(*args, **kwargs):
 def is_arraylike(x):
     """Is this object a numpy array or something similar?
 
+    This function tests specifically for an object that already has
+    array attributes (e.g. np.ndarray, dask.array.Array), **NOT** for
```

Comment:
We might add `cupy.ndarray` and `sparse.COO` arrays as two additional examples to broaden readers context beyond NumPy